### PR TITLE
Added simple string escape.

### PR DESCRIPTION
**Pull request by cseed**
Need for Seqr.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala [hunk 1]
```diff
@@ -8,6 +8,37 @@ object StringEscapeUtils {
 
   def hex(ch: Char): String = Integer.toHexString(ch).toUpperCase(Locale.ENGLISH)
 
+  def escapeStringSimple(str: String, escapeChar: Char, escape: (Char) => Boolean): String = {
+    val sb = new StringBuilder
+    str.flatMap { c =>
+      if (escape(c)) {
```
Comment (danking) [inline src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala]: We need to escape the escape character as well, right?

File: src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala [hunk 2]
```diff
@@ -8,6 +8,37 @@ object StringEscapeUtils {
...
+        val i = c.toInt
+        assert(i >= 0 && i < 256)
+        val h = Integer.toHexString(i)
+        assert(h.length == 2)
+        sb.append(h)
```
Comment (danking) [inline src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala]: Shouldn't we insert `escapeChar` before `h`?

File: src/test/scala/org/broadinstitute/hail/methods/ExprSuite.scala [hunk 3]
```diff
@@ -525,6 +525,14 @@ class ExprSuite extends SparkSuite {
...
   }
 
+  @Test def testEscapingSimple() {
+    val p = forAll { (s: String) =>
+      s == unescapeStringSimple(escapeStringSimple(s, '_', c => c.isLetterOrDigit), '_')
+    }
+    
+    p.check()
+  }
+
```
Comment (danking) [inline src/test/scala/org/broadinstitute/hail/methods/ExprSuite.scala]: Let's add a test case that acts as an example of usage:

```scala
@Test def testEscapeOnConcreteString() {
  assert(escapeStringSimple("FOO", '_', c => false) == "FOO")
  assert(escapeStringSimple("FOO", '_', c => "F") == "_"+Integer.hexString("F".toInt)+"OO")
  assert(escapeStringSimple("FOO_", '_', c => false) == "FOO_"+Integer.hexString("_".toInt))
  assert(unescapeStringSimple("FOO", '_') == "FOO")
  assert(unescapeStringSimple("FOO_"+Integer.hexString("_".toInt), '_') == "FOO_")
  assert(unescapeStringSimple("_"+Integer.hexString("F".toInt)+"OO", '_') == "FOO")
}
```

File: src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala [hunk 4]
```diff
@@ -8,6 +8,37 @@ object StringEscapeUtils {
 
   def hex(ch: Char): String = Integer.toHexString(ch).toUpperCase(Locale.ENGLISH)
 
+  def escapeStringSimple(str: String, escapeChar: Char, escape: (Char) => Boolean): String = {
+    val sb = new StringBuilder
+    str.flatMap { c =>
```
Comment (danking) [inline src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala]: I have to run, but I'm curious how this deals with surrogate pairs encoded in UTF16 which usually yield two distinct `Char`s.

File: src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala [hunk 5]
```diff
@@ -8,6 +8,37 @@ object StringEscapeUtils {
...
+    sb.result()
+  }
+
+  def unescapeStringSimple(str: String, escapeChar: Char): String = {
+    val sb = new StringBuilder
+    var i: Int = 0
+    while (i < str.length) {
+      val c = str(i)
+      if (c == escapeChar) {
+        Integer.parseInt(str.substring(i + 1, i + 3), 16)
```
Comment (danking) [inline src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala]: We ignore the result of this

**Comment by cseed** [comment]
Making additional seqr changes.  Closing for now.

**Pull request closed without merging**